marklogic-langchain4j/src/main/java/com/marklogic/langchain4j/splitter/JsonChunkDocumentProducer.java

```diff
@@ -80,4 +84,24 @@ protected DocumentWriteOperation makeChunkDocument() {
     private String determineChunksArrayName(ObjectNode doc) {
         return doc.has(DEFAULT_CHUNKS_ARRAY_NAME) ? "splitter-chunks" : DEFAULT_CHUNKS_ARRAY_NAME;
     }
+
+    private ArrayNode generateConceptsForChunk(String text) {
+        // TODO - How should we handle these exceptions?
+        try {
+            Map<String, Collection<ClassificationScore>> classificationScores = chunkConfig.getClassifier().classifyText(text);
+            ArrayNode conceptsArray = objectMapper.createArrayNode();
+            for (Map.Entry<String, Collection<ClassificationScore>> entry : classificationScores.entrySet()) {
+                for (ClassificationScore classificationScore : classificationScores.get(entry.getKey())) {
+                    ObjectNode concept = conceptsArray.addObject();
+                    concept.put("id", classificationScore.getId());
+                    concept.put("name", entry.getKey());
+                    concept.put("value", classificationScore.getName());
+                    concept.put("score", classificationScore.getScore());
+                }
+            }
+            return conceptsArray;
+        } catch (ClassificationException e) {
+            throw new RuntimeException(e);
```

Comment:
If Sonar complains about this - I think it will - change it to a ConnectorException. We'll improve error handling later.